### ISSUE TITLE
clear_host_errors not working

### ISSUE THREAD

**Issue by mikekrasn** [ansible/ansible#26227]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

`plugins/strategy/__init__.py`
##### ANSIBLE VERSION

```
ansible 2.4.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### CONFIGURATION

Fresh install

##### OS / ENVIRONMENT

RHEL
##### SUMMARY

`meta: clear_host_errors`   is not working as expected
##### STEPS TO REPRODUCE

```yaml
- hosts: all
  gather_facts: no
  tasks:

  - name: dummy check 1
    changed_when: no
    failed_when: no
    ignore_errors: yes
    shell: uname

  - meta: clear_host_errors

  - name: dummy check 2
    changed_when: no
    failed_when: no
    ignore_errors: yes
    shell: uname -n

  - meta: clear_host_errors

  - name: dummy check 3
    changed_when: no
    failed_when: no
    ignore_errors: yes
    shell: uname -n

```

##### EXPECTED RESULTS

```
$ ansible-playbook -i localhost,google.com  dummy.yml

PLAY [all] *********************************************************************

TASK [dummy check 1] ***********************************************************
ok: [localhost]
fatal: [google.com]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: ssh: connect to host google.com port 22: Connection timed out\r\n", "unreachable": true}

TASK [dummy check 2] ***********************************************************
ok: [localhost]
fatal: [google.com]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: ssh: connect to host google.com port 22: Connection timed out\r\n", "unreachable": true}

TASK [dummy check 3] ***********************************************************
ok: [localhost]
fatal: [google.com]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: ssh: connect to host google.com port 22: Connection timed out\r\n", "unreachable": true}

PLAY RECAP *********************************************************************
google.com                 : ok=0    changed=0    unreachable=3    failed=0
localhost                  : ok=2    changed=0    unreachable=0    failed=0
```
##### ACTUAL RESULTS

Expected to use unreachable host in the subsequent task. As per  https://github.com/ansible/ansible/pull/18127/files 
 "Connection failures set hosts as 'UNREACHABLE', which will remove them from the list of active hosts for the run. To recover from these issues you can use `meta: clear_host_errors` to have all currently flagged hosts reactivated, so subsequent tasks can try to use them again."

```
$ ansible-playbook -i localhost,google.com  dummy.yml

PLAY [all] *********************************************************************

TASK [dummy check 1] ***********************************************************
ok: [localhost]
fatal: [google.com]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: ssh: connect to host google.com port 22: Connection timed out\r\n", "unreachable": true}

TASK [dummy check 2] ************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [dummy check 3] ************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [dummy check 3] ************************************************************************************************************************************************************************************************************************
fatal: [google.com]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: ssh: connect to host google.com port 22: Network is unreachable\r\n", "unreachable": true}
        to retry, use: --limit @/root/dummy.retry

PLAY RECAP *********************************************************************
google.com                 : ok=0    changed=0    unreachable=2    failed=0
localhost                  : ok=3    changed=0    unreachable=0    failed=0

```

**Comment by bcoca** [comment]
confirmed, used tests, even adding a debug task, i see the 2 hosts in that but seem to 'loose the host' in the iteration right after the meta task, but regain it afterwards.

**Comment by jimi-c** [comment]
Hrm, I don't believe I had originally meant this to clear unreachable errors, which we consider a different class of error from failed tasks. I'm not sure if we want to fix this to do that as well, or if we want to add a new meta specifically for unreachable hosts.

**Comment by promaty** [comment]
@jimi-c, please add a way to handle unreachable errors, this would help immensely (either through meta method or something else).

**Comment by ghost** [comment]
@jimi-c and @promaty, I agree handling unreachable error programmatically would be a great enhancement.

**Comment by PisikeSipelgas** [comment]
obviously ansible 2.5.2 has the same problem.

**Comment by PisikeSipelgas** [comment]
What's the status of this one? This issue still persists in ansible 2.5.5.

**Comment by PisikeSipelgas** [comment]
Sorry to bother again, but is there a real milestone for this? 
With Ansible 2.6.2 is still not working as expected.

**Comment by bcoca** [comment]
@PisikeSipelgas there has been no development on this yet. It is something we want to get to, but it is not probable that a core team member can get to this soon.

**Comment by kinwolfqc** [comment]
Status on this?  Still not fixed in 2.8 and yet the doc is pretty clear that using `meta: clear_host_errors` is the method that should be used to clear unreachable host error.

https://docs.ansible.com/ansible/latest/user_guide/playbooks_error_handling.html#resetting-unreachable-hosts

**Comment by kinwolfqc** [comment]
Further search shows that this issue might be closed but the doc needs updating to reflect that there is a new keyword `ignore_unreachable` since v2.7.  I tested it and it works.
